--- a/net/minecraft/world/entity/animal/coppergolem/CopperGolemAi.java
+++ b/net/minecraft/world/entity/animal/coppergolem/CopperGolemAi.java
@@ -43,7 +_,7 @@
     private static final int TICK_TO_START_ON_REACHED_INTERACTION = 1;
     private static final int TICK_TO_PLAY_ON_REACHED_SOUND = 9;
     private static final Predicate<BlockState> TRANSPORT_ITEM_SOURCE_BLOCK = blockState -> blockState.is(BlockTags.COPPER_CHESTS);
-    private static final Predicate<BlockState> TRANSPORT_ITEM_DESTINATION_BLOCK = blockState -> blockState.is(Blocks.CHEST)
+    private static final Predicate<BlockState> TRANSPORT_ITEM_DESTINATION_BLOCK = blockState -> blockState.is(Blocks.CHEST) // Purpur - copper golem can place items in barrels option - diff on change
         || blockState.is(Blocks.TRAPPED_CHEST);
     private static final ImmutableList<SensorType<? extends Sensor<? super CopperGolem>>> SENSOR_TYPES = ImmutableList.of(
         SensorType.NEAREST_LIVING_ENTITIES, SensorType.HURT_BY
