From 73b67b6fc4ab1a46cb7c1113a4648915bf072f56 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 05:37:42 -0800
Subject: [PATCH] purpur File Patches


diff --git a/net/minecraft/world/item/ProjectileWeaponItem.java b/net/minecraft/world/item/ProjectileWeaponItem.java
index 663826d7e26bf647376eacbf0c6a2bb80bbdf55a..aa1cf1653d33cd35f77db28dbee0c2319e1975f3 100644
--- a/net/minecraft/world/item/ProjectileWeaponItem.java
+++ b/net/minecraft/world/item/ProjectileWeaponItem.java
@@ -106,6 +106,8 @@ public abstract class ProjectileWeaponItem extends Item {
             abstractArrow.setCritArrow(true);
         }
 
+        abstractArrow.setActualEnchantments(weapon.getEnchantments()); // Purpur - Add an option to fix MC-3304 projectile looting
+
         return abstractArrow;
     }
 
