From 1e4693c6129676cacc2e29a157b5a69107c78543 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 06:37:42 -0700
Subject: [PATCH] purpur File Patches


diff --git a/net/minecraft/world/entity/monster/ZombifiedPiglin.java b/net/minecraft/world/entity/monster/ZombifiedPiglin.java
index 822712eaff2f6c579d982734ab14a00c02182770..1be7c756d56cca63347d97f7896b491f3a343fa4 100644
--- a/net/minecraft/world/entity/monster/ZombifiedPiglin.java
+++ b/net/minecraft/world/entity/monster/ZombifiedPiglin.java
@@ -113,6 +113,12 @@ public class ZombifiedPiglin extends Zombie implements NeutralMob {
             this.maybeAlertOthers();
         }
 
+        // Purpur start - Toggle for Zombified Piglin death always counting as player kill when angry
+        if (this.isAngry() && this.level().purpurConfig.zombifiedPiglinCountAsPlayerKillWhenAngry) {
+            this.lastHurtByPlayerMemoryTime = this.tickCount;
+        }
+        // Purpur end - Toggle for Zombified Piglin death always counting as player kill when angry
+
         super.customServerAiStep(level);
     }
 
@@ -160,6 +166,12 @@ public class ZombifiedPiglin extends Zombie implements NeutralMob {
             this.ticksUntilNextAlert = ALERT_INTERVAL.sample(this.random);
         }
 
+        // Purpur start - Toggle for Zombified Piglin death always counting as player kill when angry
+        if (livingEntity instanceof Player player && this.level().purpurConfig.zombifiedPiglinCountAsPlayerKillWhenAngry) {
+            this.setLastHurtByPlayer(player, this.tickCount);
+        }
+        // Purpur end - Toggle for Zombified Piglin death always counting as player kill when angry
+
         return super.setTarget(livingEntity, reason); // CraftBukkit
     }
 
