From 73b67b6fc4ab1a46cb7c1113a4648915bf072f56 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 05:37:42 -0800
Subject: [PATCH] purpur File Patches


diff --git a/net/minecraft/world/entity/animal/MushroomCow.java b/net/minecraft/world/entity/animal/MushroomCow.java
index 7b4f76ae8907ae45cb0ac6fa397a319ab3cc2a22..3ce300a6abc4ee39f010ff5318b64c50e732742b 100644
--- a/net/minecraft/world/entity/animal/MushroomCow.java
+++ b/net/minecraft/world/entity/animal/MushroomCow.java
@@ -201,6 +201,13 @@ public class MushroomCow extends AbstractCow implements Shearable {
         level.playSound(null, this, SoundEvents.MOOSHROOM_SHEAR, source, 1.0F, 1.0F);
         this.convertTo(EntityType.COW, ConversionParams.single(this, false, false), cow -> {
             level.sendParticles(ParticleTypes.EXPLOSION, this.getX(), this.getY(0.5), this.getZ(), 1, 0.0, 0.0, 0.0, 0.0);
+            // Purpur start - Fix cow rotation when shearing mooshroom
+            cow.copyPosition(this);
+            cow.yBodyRot = this.yBodyRot;
+            cow.setYHeadRot(this.getYHeadRot());
+            cow.yRotO = this.yRotO;
+            cow.xRotO = this.xRotO;
+            // Purpur end - Fix cow rotation when shearing mooshroom
             // Paper start - custom shear drops; moved drop generation to separate method
             drops.forEach(drop -> {
                 this.spawnAtLocation(level, new ItemEntity(this.level(), this.getX(), this.getY(1.0), this.getZ(), drop));
