--- a/src/main/java/org/spigotmc/SpigotConfig.java
+++ b/src/main/java/org/spigotmc/SpigotConfig.java
@@ -68,13 +_,12 @@
         SpigotConfig.config.options().copyDefaults(true);
 
         SpigotConfig.commands = new HashMap<>();
-        SpigotConfig.commands.put("spigot", new SpigotCommand("spigot"));
 
         SpigotConfig.version = SpigotConfig.getInt("config-version", 13);
         SpigotConfig.set("config-version", 13);
         SpigotConfig.readConfig(SpigotConfig.class, null);
     }
 
     public static void registerCommands() {
         for (Map.Entry<String, Command> entry : SpigotConfig.commands.entrySet()) {
-            MinecraftServer.getServer().server.getCommandMap().register(entry.getKey(), "Spigot", entry.getValue());
+            MinecraftServer.getServer().server.getCommandMap().register(entry.getKey(), "honor", entry.getValue());
